tests/unit_tests/commandline/test_geoips_run.py

```diff
@@ -0,0 +1,124 @@
+"""Unit test for GeoIPS CLI `run` command.
+
+See geoips/commandline/ancillary_info/cmd_instructions.yaml for more information.
+"""
+
+from glob import glob
+from importlib import resources
+from numpy.random import rand
+from os import listdir
+from os.path import basename
+import pytest
+
+from tests.unit_tests.commandline.cli_top_level_tester import BaseCliTest
+
+
+class TestGeoipsRun(BaseCliTest):
+    """Unit Testing Class for GeoipsRun Command."""
+
+    @property
+    def all_possible_subcommand_combinations(self):
+        """A list of every possible call signature for the GeoipsRun command.
+
+        This includes failing cases as well.
+        """
+        if not hasattr(self, "_cmd_list"):
+            self._cmd_list = []
+            base_args = self._run_args
+            test_data_dir = str(resources.files("geoips") / "../../test_data")
+            # select a small random amount of tests to call via geoips run
+            for pkg_name in self.plugin_packages:
+                script_paths = sorted(
+                    [
+                        script_path
+                        for script_path in glob(
+                            str(resources.files(pkg_name) / "../tests/scripts/*.sh")
+                        )
+                    ]
+                )
+                for script_path in script_paths:
+                    do_geoips_run = rand() < 0.15
```

Comment:
perhaps we include an option to run all of these OR a random subset? See above comments as well

Reply:
See opened issue for this